### PR TITLE
Added transfer api to REST endpoint

### PR DESCRIPTION
**Pull request by MartinSchmidt**
Added REST api to queue a transfer request of a certificate to another wallet for the authenticated user.

### CODE REVIEW

File: src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs [hunk 1]
```diff
@@ -53,7 +54,7 @@ public async Task Consume(ConsumeContext<TransferCertificateCommand> context)
             List<Task> tasks = new();
             foreach (var slice in reservedSlices)
             {
-                var builder = new RoutingSlipBuilder(NewId.NextGuid());
+                var builder = new RoutingSlipBuilder(msg.TransferRequestId);
```
Comment (tnickelsen) [inline src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs:57]: Why this change?

File: src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/TransfersController.cs [hunk 2]
```diff
@@ -100,4 +102,117 @@ public async Task<ActionResult<ResultList<Transfer>>> GetTransfers([FromServices
...
+            TransferRequestId = Guid.NewGuid(),
+            Owner = subject,
+            Registry = request.CertificateId.Registry,
+            CertificateId = request.CertificateId.StreamId,
+            Quantity = request.Quantity,
+            Receiver = request.ReceiverId,
+            HashedAttributes = request.HashedAttributes,
+        };
+
+        await bus.Publish(command);
```
Comment (tnickelsen) [inline src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/TransfersController.cs:136]: I don't believe this should be awaited?

File: src/Protos/wallet.proto [hunk 3]
```diff
@@ -84,7 +84,9 @@ message TransferRequest {
     repeated string hashed_attributes = 4; // the hashed attribute keys to transfer
 }
 
-message TransferResponse {} // not yet sure what to return
+message TransferResponse {
+    project_origin.common.v1.Uuid transfer_id = 1;
```
Comment (tnickelsen) [inline src/Protos/wallet.proto:88]: Why this change?

File: src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/TransfersController.cs (hunk 2 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/TransfersController.cs:136]: It should, to make sure it happens on the same task and before the response is returns to the caller.
Otherwise it could fail in a separate task.

File: src/Protos/wallet.proto (hunk 3 above)
Comment (MartinSchmidt) [reply src/Protos/wallet.proto:88]: To make it on par with the new rest api, and claim api, where a ClaimId is also returns, it is so we later can query back on the task started to get a state of it if wanted.

File: src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs (hunk 1 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.Server/CommandHandlers/TransferCertificateCommandHandler.cs:57]: Since the ID is now generated earlier in the flow, in the controller, so it can be returned to the caller.

File: src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/TransfersController.cs (hunk 2 above)
Comment (tnickelsen) [reply src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/TransfersController.cs:136]: Maybe we should talk about this over teams

**Review by tnickelsen — APPROVED** [review]
